strictdoc/export/html/generators/view_objects/document_screen_view_object.py

```diff
@@ -233,41 +240,43 @@ def render_standalone_document_link(
             raise NotImplementedError
         root_prefix = document.meta.get_root_path_prefix()
         document_link = document.meta.get_html_standalone_document_link()
-        if len(root_prefix) == 0:
-            return document_link
-        return "/".join((root_prefix, document_link))
+        return (
+            document_link
+            if len(root_prefix) == 0
+            else "/".join((root_prefix, document_link))
+        )
 
-    def render_static_url(self, url: str):
-        return self.link_renderer.render_static_url(url)
+    def render_static_url(self, url: str) -> str:
```

Comment:
Why not `-> Markup` here?

Reply:
It was an oversight. `-> str` is correct. Wrapping in `Markup()` is wrong. Reasoning is basically the same as in PR description, "Considerations about what to escape - render_*_link make...".

I intended to review and annotate all `strictdoc/export/html/generators/view_objects/*.py` for `str` vs. `Markup`, but was too slow. Another PR will follow.

Thanks for merging this early anyway. We use it for daily work now and experienced now failures so far.

Reply:
Why would wrapping into `Markup` would be wrong in this case? Isn't it a string that doesn't have to be escaped?

Reply:
> Isn't it a string that doesn't have to be escaped?

The string is used in places like `<link href="the_string">`. It usually contains the name of a file. There are two scenarios to consider:
- Non malicious: Let the filename contain a whitespace, e.g. `"some file.css"`. We end up with `<link href="some file.css">`, which is not a valid URL. The string must be URL escaped. `<link href="some%20file.css">` would be correct.
- Malicious: The filename is crafted so that causes injection. Let it be `'somefile.css"><script>bad()</script><div id="a` and we end up with `<link href="somefile.css"><script>bad()</script><div id="a">`.

One could argue the input string is trusted because it'll only ever contain hardcoded static file names and so the 2nd scenario can't happen. Then HTML escaping would not be necessary.

We should still consider URL escaping, which I suggested to do in a future MR due to current lack of time.

Reply:
This makes sense. I didn't consider this escaping topic strongly before, and your work helps me understand this better and is moving the project to a very good direction.

It would be great if you could include this your comment to the design document as well. It helps a lot in understanding what and why we are doing with escaping. Also, this all contributes to at least the basic handling of the security topic by StrictDoc.